backend/inspirehep/records/marshmallow/authors/base.py

```diff
@@ -16,6 +21,32 @@ class AuthorsRawSchema(RecordBaseSchema):
     advisors = NonHiddenRaw(dump_only=True)
     project_membership = NonHiddenRaw(dump_only=True)
 
+    @staticmethod
+    def get_student_name(data):
+        preferred_name = get_value(data, "name.preferred_name")
+        if not preferred_name:
+            preferred_name = get_value(data, "name.value")
+        return preferred_name
+
+    def populate_students_field(self, record):
+        student_records = (
+            db.session.query(RecordMetadata.json, StudentsAdvisors.degree_type)
+            .filter(
+                StudentsAdvisors.student_id == RecordMetadata.id,
+                StudentsAdvisors.advisor_id == record.id,
+            )
+            .all()
+        )
+        student_entries = []
+        for student_json, student_degree in student_records:
+            student_data = {
+                "name": self.get_student_name(student_json),
```

Comment:
wouldn't it be better to store this data in the new table too, instead of doing the join here?

Reply:
I think that this should be quite fast since we have index on `student_id`

Reply:
IMO it's better to use a simple join instead of duplicating data